internal/operations/manager.go

```diff
@@ -77,6 +78,18 @@ type operationsManager struct {
 	cache     cache.CInterface
 }
 
+// SubmitBulkOperationUpdate implements Manager.
+func (om *operationsManager) SubmitBulkOperationUpdates(ctx context.Context, updates []*core.OperationUpdate, onCommit chan<- error) {
+	for _, update := range updates {
+		errString := ""
+		if update.ErrorMessage != "" {
+			errString = fmt.Sprintf(" error=%s", update.ErrorMessage)
+		}
+		log.L(om.ctx).Debugf("%s updating operation %s status=%s%s", update.Plugin, update.NamespacedOpID, update.Status, errString)
+	}
+	om.updater.SubmitBulkOperationUpdates(ctx, updates, onCommit)
```

Comment:
What is it that absolutely guarantees that this list of receipts in the order they were submitted, will be committed before any other receipts submitted to this function?

Reply:
👍 - with the other discussions to make it a synchronous interface, I think you could just handle this as a comment on the API doc:

> An insertion ordering guarantee is only provided when this code is called on a single goroutine inside of the connector. It is the responsibility of the connector code to allocate that routine, and ensure that there is only one.